lib/std/math/nextafter.zig

```diff
@@ -0,0 +1,329 @@
+    try expect(nextAfter(f32, -0x1p-126, math.inf(f32)) == -0x0.FFFFFEp-126);
+    try expect(nextAfter(f64, 0x1p-1022, -math.inf(f64)) == 0x0.FFFFFFFFFFFFFp-1022);
+    try expect(nextAfter(f64, -0x1p-1022, math.inf(f64)) == -0x0.FFFFFFFFFFFFFp-1022);
+    try expect(nextAfter(f128, 0x1p-16382, -math.inf(f128)) == 0x0.FFFFFFFFFFFFFFFFFFFFFFFFFFFFp-16382);
+    try expect(nextAfter(f128, -0x1p-16382, math.inf(f128)) == -0x0.FFFFFFFFFFFFFFFFFFFFFFFFFFFFp-16382);
+
+    // subnormal -> normal
+    try expect(nextAfter(f16, 0x0.FFCp-14, math.inf(f16)) == 0x1p-14);
+    try expect(nextAfter(f16, -0x0.FFCp-14, -math.inf(f16)) == -0x1p-14);
+    try expect(nextAfter(f32, 0x0.FFFFFEp-126, math.inf(f32)) == 0x1p-126);
+    try expect(nextAfter(f32, -0x0.FFFFFEp-126, -math.inf(f32)) == -0x1p-126);
+    try expect(nextAfter(f64, 0x0.FFFFFFFFFFFFFp-1022, math.inf(f64)) == 0x1p-1022);
+    try expect(nextAfter(f64, -0x0.FFFFFFFFFFFFFp-1022, -math.inf(f64)) == -0x1p-1022);
+    try expect(nextAfter(f128, 0x0.FFFFFFFFFFFFFFFFFFFFFFFFFFFFp-16382, math.inf(f128)) == 0x1p-16382);
+    try expect(nextAfter(f128, -0x0.FFFFFFFFFFFFFFFFFFFFFFFFFFFFp-16382, -math.inf(f128)) == -0x1p-16382);
+
+    // special values
+    inline for (.{ f16, f32, f64, f128 }) |T| {
+        try expect(bitwiseEqual(T, nextAfter(T, 0.0, 0.0), 0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, 0.0, -0.0), -0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, -0.0, -0.0), -0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, -0.0, 0.0), 0.0));
+        try expect(nextAfter(T, 0.0, math.inf(T)) == math.floatTrueMin(T));
+        try expect(nextAfter(T, 0.0, -math.inf(T)) == -math.floatTrueMin(T));
+        try expect(nextAfter(T, -0.0, -math.inf(T)) == -math.floatTrueMin(T));
+        try expect(nextAfter(T, -0.0, math.inf(T)) == math.floatTrueMin(T));
+        try expect(bitwiseEqual(T, nextAfter(T, math.floatTrueMin(T), 0.0), 0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, math.floatTrueMin(T), -0.0), 0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, math.floatTrueMin(T), -math.inf(T)), 0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, -math.floatTrueMin(T), -0.0), -0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, -math.floatTrueMin(T), 0.0), -0.0));
+        try expect(bitwiseEqual(T, nextAfter(T, -math.floatTrueMin(T), math.inf(T)), -0.0));
+        try expect(nextAfter(T, math.inf(T), math.inf(T)) == math.inf(T));
+        try expect(nextAfter(T, math.inf(T), -math.inf(T)) == math.floatMax(T));
+        try expect(nextAfter(T, math.floatMax(T), math.inf(T)) == math.inf(T));
+        try expect(nextAfter(T, -math.inf(T), -math.inf(T)) == -math.inf(T));
+        try expect(nextAfter(T, -math.inf(T), math.inf(T)) == -math.floatMax(T));
+        try expect(nextAfter(T, -math.floatMax(T), -math.inf(T)) == -math.inf(T));
+        try expect(math.isNan(nextAfter(T, 1.0, math.nan(T))));
```

Comment:
To make things complete, add +-inf -> nan and vice versa.

Reply:
Added.